File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    //prompt for input
    string text = get_string("Text: ");

    //number of letters
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
    }

    //numbers of words
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }

    //numbers of sentences
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    //apply formula
    float calculation = (0.0588 * letters / words * 100) - (0.296 * sentences / words * 100) - 15.8;

    int index = round(calculation);

    //result
    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else
    {
        printf("Grade %i\n", index);
        return 0;
    }
}

File: sentimental-readability/readability.py
# Todo
import cs50

# Prompt user for input
    # get_string
text = cs50.get_string("Text: ")

# Defining
words = 1
letters = 0
sentences = 0

# Iterate:
for i in text:

    # Number of words
    if i == " ":
        words += 1

    # Number of letters
    elif i.isalpha():
        letters += 1

    # Number of sentences
    elif i == "." or i == "!" or i == "?":
        sentences += 1

# Apply formula
    # 0.0588 * L - 0.296 * S - 15.8
index = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8
index = round(index)

# Result
    # If index < 1: Before Grade 1
if index < 1:
    print("Before Grade 1")

    # If index > 16: Grade 16+
elif index >= 16:
    print("Grade 16+")

    # Else: Grade {index}
else:
    print(f"Grade {index}")